be/src/exec/es/es_predicate.cpp

```diff
@@ -0,0 +1,419 @@
+            expr = conjunct->get_child(0);
+            slotRef = (SlotRef*)(conjunct->get_child(1));
+        } else {
+            VLOG(1) << "get disjuncts fail: no SLOT_REF child";
+            return false;
+        }
+
+        const SlotDescriptor* slot_desc = get_slot_desc(slotRef);
+        if (slot_desc == nullptr) {
+            VLOG(1) << "get disjuncts fail: slot_desc is null";
+            return false;
+        }
+
+        PrimitiveType type = expr->type().type;
+        if (type != TYPE_VARCHAR && type != TYPE_CHAR) {
+            VLOG(1) << "get disjuncts fail: like value is not a string";
+            return false;
+        }
+
+        ExtLiteral literal(type, _context->get_value(expr, NULL));
+        ExtPredicate* predicate = new ExtLikePredicate(
+                    TExprNodeType::LIKE_PRED,
+                    slot_desc->col_name(),
+                    slot_desc->type(),
+                    literal);
+
+        _disjuncts.push_back(predicate);
+        return true;
+    }
+      
+    if (TExprNodeType::IN_PRED == conjunct->node_type()) {
+        // the op code maybe FILTER_NEW_IN, it means there is function in list
+        // like col_a in (abs(1))
+        if (TExprOpcode::FILTER_IN != conjunct->op() 
+            && TExprOpcode::FILTER_NOT_IN != conjunct->op()) {
+            return false;
+        }
+
+        vector<ExtLiteral> in_pred_values;
+        const InPredicate* pred = dynamic_cast<const InPredicate*>(conjunct);
+        if (Expr::type_without_cast(pred->get_child(0)) != TExprNodeType::SLOT_REF) {
+            return false;
+        }
+
+        SlotRef* slot_ref = (SlotRef*)(conjunct->get_child(0));
+        const SlotDescriptor* slot_desc = get_slot_desc(slot_ref);
+        if (slot_desc == nullptr) {
+            return false;
+        }
+
+        if (pred->get_child(0)->type().type != slot_desc->type().type) {
+            if (!ignore_cast(slot_desc, pred->get_child(0))) {
+                return false;
+            }
+        }
+
+        HybirdSetBase::IteratorBase* iter = pred->hybird_set()->begin();
+        while (iter->has_next()) {
+            if (nullptr == iter->get_value()) {
+                return false;
+            }
+
+            ExtLiteral literal(slot_desc->type().type, const_cast<void *>(iter->get_value()));
+            in_pred_values.emplace_back(literal);
+            iter->next();
+        }
+
+        ExtPredicate* predicate = new ExtInPredicate(
+                    TExprNodeType::IN_PRED,
+                    pred->is_not_in(),
+                    slot_desc->col_name(),
+                    slot_desc->type(),
+                    in_pred_values);
+        _disjuncts.push_back(predicate);
+
+        return true;
+    } 
+    
+    if (TExprNodeType::COMPOUND_PRED == conjunct->node_type()) {
+        if (TExprOpcode::COMPOUND_OR != conjunct->op()) {
+            VLOG(1) << "get disjuncts fail: op is not COMPOUND_OR";
```

Comment:
The conjunct list BE received from FE is Conjunctive normal form, so the relationship between each element is `and`, the disconjunctive normal form is in single element.  it can't express OR relationship right, and can't handle conjuenct containing `AND` and `OR`.

Reply:
Yes, the conjunct here does not been push down to ES now, we will improve it by push such expression to ES in future.